test/utils/setup.ts

```diff
@@ -118,21 +130,23 @@ export const getSafeTemplate = async (saltNumber: string = getRandomIntAsString(
     const factory = await getFactory();
     const template = await factory.createProxyWithNonce.staticCall(singletonAddress, "0x", saltNumber);
     await factory.createProxyWithNonce(singletonAddress, "0x", saltNumber).then((tx: any) => tx.wait());
-    const Safe = await getSafeSingletonContractFromEnvVariable();
+    const Safe = await getSafeSingletonContractFactoryFromEnvVariable();
     return Safe.attach(template) as Safe | SafeL2;
 };
 
 export const getSafeWithOwners = async (
     owners: string[],
-    threshold?: number,
-    fallbackHandler?: string,
-    logGasUsage?: boolean,
+    threshold: number = owners.length,
+    to: string = AddressZero,
+    data: string = "0x",
```

Comment:
Previously, our Safe setup function didn't allow specifying the optional call, so I had to refactor it a bit, including the usage. If we decide to use the SafeToL2Setup contract by default, this will be handy.

Reply:
Maybe provide a separate `getSafeWithOwnersAndSetup` to avoid so many changes?

Reply:
replied here https://github.com/safe-global/safe-smart-account/pull/759#discussion_r1663853026